src/CMakeLists.txt

```diff
@@ -340,18 +332,16 @@ target_link_libraries(wesnoth-sdl
 
 ########### Wesnoth main source files ###############
 
-# Depending on whether the game or the game and the tests are build the main
-# sources are compiled in the game library or not. If only one target is needed
-# the putting in the game library solution is slower. If both targets are build
-# the library is faster (especially if the user doesn't use ccache (or
-# something similar)).
+# create libwesnoth-extras.a if building the client or the boost unit tests
 
-GetSources("wesnoth" wesnoth-main_SRC)
+if(ENABLE_GAME OR ENABLE_TESTS)
+	GetSources("libwesnoth_extras" wesnoth-main_SRC)
+	add_library(wesnoth-extras ${LIBRARY_TYPE} EXCLUDE_FROM_ALL ${wesnoth-main_SRC})
+endif(ENABLE_GAME OR ENABLE_TESTS)
 
 ########### libwesnoth-game ###############
-# rename libwesnoth.a to libwesnoth-game.a to have clearer targets
 
-GetSources("libwesnoth" libwesnoth-game_STAT_SRC)
+GetSources("libwesnoth_game" libwesnoth-game_STAT_SRC)
```

Comment:
Likewise, here I prefer keeping the file name in source_lists as "libwesnoth" instead of "libwesnoth_game".

Reply:
Alright.  In scons and cmake I would prefer to keep the "game" suffix for the variables though - it makes them a lot easier to search on.